### PR TITLE
Build tweaks

### PR DESCRIPTION
**Pull request by benfoxall**
Getting the site running locally can be kind of difficult.  Also, debugging install issues is harder because they are hidden behind npm install scripts.

This PR:
* strips back the required gems
* separates out the build steps so that the site can be served without the optimisation dependencies

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,46 +3,80 @@ jsoxford.github.com
...
 ```
 
-If you're feeling really adventurous, you can deploy it like so:
+You should now be able to visit http://localhost:4000 and see the JSOxford site.
+
+### 3. Build an optimised version of the site
+
+```bash
+# get the gulp task runner
+npm install -g gulp-cli
```
Comment (danielthepope) [inline README.md]: Doesn't this site use grunt, not gulp?

Comment (benfoxall) [inline README.md]: YES!

**Comment by AverageMarcus** [comment]
Still fails trying to install `charlock_holmes (0.7.3)` with 
```
***************************************************************************************
*********** icu required (brew install icu4c or apt-get install libicu-dev) ***********
***************************************************************************************
```

**Comment by benfoxall** [comment]
Weird - `charlock_holmes` is no longer in the [Gemfile.lock](https://github.com/jsoxford/jsoxford.github.com/commit/b00dee72c5626b53ea1ab9d054cd785d6b23f096#diff-e79a60dc6b85309ae70a6ea8261eaf95)..  hmm

**Comment by benfoxall** [comment]
Would you be able to pop the entire bundler output somewhere? (here's [mine](https://gist.github.com/benfoxall/028d722706168e9d1f0c))

**Comment by AverageMarcus** [comment]
[Here you go](https://gist.github.com/AverageMarcus/03cf1f02626f98412a5a) @benfoxall

**Comment by AverageMarcus** [comment]
Ugh! I'm an idiot! Was on the wrong branch!

I do still get an error fetching gems `Gem::RemoteFetcher::FetchError: SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed (https://rubygems.org/gems/coffee-script-source-1.9.1.1.gem)` but changing the first line of the Gemfile to `source 'http://rubygems.org'` ~~fixes~~ works around this for me.

**Pull request merged**